### ISSUE TITLE
Bug - Recipe step for inner recipes - execution id in json is being ignored in code

### ISSUE THREAD

**Issue by JoshTango** [OrchardCMS/OrchardCore#13132]
#13094 as per this issue @jtkech discovered that the execution id in the recipes step for inner recipes was not being used internally in the code.  The execution id was being ignored and set to a new guid.

So I guess decide either to remove the execution id in the json file for inner recipes or to fix the issue so that the declared execution id is used in the code.

Its up to you.  Just bring this to your attention.

**Comment by sebastienros** [comment]
@jtkech can you confirm it's a valid issue that needs to be handled?

**Comment by jtkech** [comment]
Yes, this defined executionId is not passed to the recipe descriptor so before executing it we generate another one, but maybe the first was intended to be used to define different groups having the same executionId, each inner recipe still needing a real and unique other executionId.

This because in `RecipesStep.cs` I saw that we populate a dictionary to groups recipes having the same defined executionId. But this seems to be useless as this dictionary is not used afterwards.

Also when looping on the defined inner recipes, I saw that we harvest all recipes on each iteration in place of doing it once before the loop.

So yes, maybe worth to review `RecipesStep.cs` and do a PR pointing to this issue.

In fact it seems that currently this executionId is useless, but we still need to define it for a inner recipe, otherwise an exception would be thrown when populating the useless dictionary with a null key ;)
